### ISSUE TITLE
Panic when reading a deleted workspace

### ISSUE THREAD

**Issue by sean-nixon** [terraform-providers/terraform-provider-tfe#149]
### Terraform (and TFE Provider) Version
```
Terraform v0.12.24
+ provider.tfe v0.15.1
```

### Affected Resource(s)

* `tfe_workspace`

### Panic Output
```
Error: rpc error: code = Unavailable desc = transport is closing


panic: runtime error: invalid memory address or nil pointer dereference
2020-03-30T12:02:15.064-0500 [DEBUG] plugin.terraform-provider-tfe_v0.15.1_x4: [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x19a8c6c]
2020-03-30T12:02:15.064-0500 [DEBUG] plugin.terraform-provider-tfe_v0.15.1_x4:
2020-03-30T12:02:15.064-0500 [DEBUG] plugin.terraform-provider-tfe_v0.15.1_x4: goroutine 163 [running]:
2020-03-30T12:02:15.064-0500 [DEBUG] plugin.terraform-provider-tfe_v0.15.1_x4: github.com/terraform-providers/terraform-provider-tfe/tfe.resourceTFEWorkspaceRead(0xc0002bb260, 0x1b43680, 0xc0004861e0, 0xc0002bb260, 0x0)
2020-03-30T12:02:15.064-0500 [DEBUG] plugin.terraform-provider-tfe_v0.15.1_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-tfe/tfe/resource_tfe_workspace.go:208 +0x2cc
2020-03-30T12:02:15.064-0500 [DEBUG] plugin.terraform-provider-tfe_v0.15.1_x4: github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Resource).RefreshWithoutUpgrade(0xc000113700, 0xc00027ef50, 0x1b43680, 0xc0004861e0, 0xc0008c0600, 0x0, 0x0)
2020-03-30T12:02:15.064-0500 [DEBUG] plugin.terraform-provider-tfe_v0.15.1_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-tfe/vendor/github.com/hashicorp/terraform-plugin-sdk/helper/schema/resource.go:455 +0x11c
2020-03-30T12:02:15.064-0500 [DEBUG] plugin.terraform-provider-tfe_v0.15.1_x4: github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin.(*GRPCProviderServer).ReadResource(0xc00000cab0, 0x1dd2b20, 0xc000452fc0, 0xc00027ee60, 0xc00000cab0, 0xc000452f60, 0x1ad31a0)
2020-03-30T12:02:15.064-0500 [DEBUG] plugin.terraform-provider-tfe_v0.15.1_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-tfe/vendor/github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin/grpc_provider.go:525 +0x3a8
2020-03-30T12:02:15.064-0500 [DEBUG] plugin.terraform-provider-tfe_v0.15.1_x4: github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5._Provider_ReadResource_Handler(0x1bde520, 0xc00000cab0, 0x1dd2b20, 0xc000452fc0, 0xc0003e7140, 0x0, 0x0, 0x0, 0xc0007801c0, 0x1a7)
2020-03-30T12:02:15.064-0500 [DEBUG] plugin.terraform-provider-tfe_v0.15.1_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-tfe/vendor/github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5/tfplugin5.pb.go:3153 +0x23e
2020-03-30T12:02:15.064-0500 [DEBUG] plugin.terraform-provider-tfe_v0.15.1_x4: google.golang.org/grpc.(*Server).processUnaryRPC(0xc0000bc000, 0x1dd95e0, 0xc00058e480, 0xc000539800, 0xc0001223f0, 0x2574270, 0x0, 0x0, 0x0)
2020-03-30T12:02:15.064-0500 [DEBUG] plugin.terraform-provider-tfe_v0.15.1_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-tfe/vendor/google.golang.org/grpc/server.go:995 +0x485
2020-03-30T12:02:15.064-0500 [DEBUG] plugin.terraform-provider-tfe_v0.15.1_x4: google.golang.org/grpc.(*Server).handleStream(0xc0000bc000, 0x1dd95e0, 0xc00058e480, 0xc000539800, 0x0)
2020-03-30T12:02:15.064-0500 [DEBUG] plugin.terraform-provider-tfe_v0.15.1_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-tfe/vendor/google.golang.org/grpc/server.go:1275 +0xe02
2020-03-30T12:02:15.064-0500 [DEBUG] plugin.terraform-provider-tfe_v0.15.1_x4: google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc000038580, 0xc0000bc000, 0x1dd95e0, 0xc00058e480, 0xc000539800)
2020-03-30T12:02:15.064-0500 [DEBUG] plugin.terraform-provider-tfe_v0.15.1_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-tfe/vendor/google.golang.org/grpc/server.go:710 +0x9f
2020-03-30T12:02:15.064-0500 [DEBUG] plugin.terraform-provider-tfe_v0.15.1_x4: created by google.golang.org/grpc.(*Server).serveStreams.func1
2020-03-30T12:02:15.064-0500 [DEBUG] plugin.terraform-provider-tfe_v0.15.1_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-tfe/vendor/google.golang.org/grpc/server.go:708 +0xa1
2020-03-30T12:02:15.066-0500 [DEBUG] plugin: plugin process exited: path=/Users/snixon/Documents/source/mrcglobal/terraform/tfe/terraform-enterprise-config/terraform_enterprise/dev/organizations/.terraform/plugins/darwin_amd64/terraform-provider-tfe_v0.15.1_x4 pid=30661 error="exit status 2"
2020/03/30 12:02:15 [ERROR] <root>: eval: *terraform.EvalRefresh, err: rpc error: code = Unavailable desc = transport is closing
2020/03/30 12:02:15 [ERROR] <root>: eval: *terraform.EvalSequence, err: rpc error: code = Unavailable desc = transport is closing
2020/03/30 12:02:15 [TRACE] [walkRefresh] Exiting eval tree: tfe_workspace.root_workspace
2020/03/30 12:02:15 [TRACE] vertex "tfe_workspace.root_workspace": visit complete
2020/03/30 12:02:15 [TRACE] vertex "tfe_workspace.root_workspace": dynamic subgraph encountered errors
2020/03/30 12:02:15 [TRACE] vertex "tfe_workspace.root_workspace": visit complete
2020/03/30 12:02:15 [TRACE] dag/walk: upstream of "tfe_variable.team_token_variables" errored, so skipping
2020/03/30 12:02:15 [TRACE] dag/walk: upstream of "tfe_variable.sensitive_variables" errored, so skipping
2020/03/30 12:02:15 [TRACE] dag/walk: upstream of "tfe_variable.vcs_token_variables" errored, so skipping
2020/03/30 12:02:15 [TRACE] dag/walk: upstream of "tfe_variable.nonsensitive_variables" errored, so skipping
2020/03/30 12:02:15 [TRACE] dag/walk: upstream of "tfe_notification_configuration.root_workspace_notifications" errored, so skipping
2020/03/30 12:02:15 [TRACE] dag/walk: upstream of "provider.tfe (close)" errored, so skipping
2020/03/30 12:02:15 [TRACE] dag/walk: upstream of "root" errored, so skipping
2020-03-30T12:02:15.311-0500 [DEBUG] plugin: plugin exited
```

### Expected Behavior

If a workspace is deleted outside of Terraform, subsequent plans/applys will detect the workspace as deleted and recreate it.

### Actual behavior

Panic while refreshing the resource.

### Steps to reproduce

* Create a workspace using the `tfe_workspace` resource
* Delete the workspace manually in Terraform Cloud/Enterprise
* Run `terraform apply`

**Comment by scottwinkler** [comment]
Same issue.

**Comment by mikemikemotorbike** [comment]
Same issue. We did notice the panic only happens when the backend is TFE.  S3 backends throw an error without panic

**Comment by limakzi** [comment]
Up-vote. It affects both `0.15.1` and `0.16.0`

**Comment by leetrout** [comment]
This is fixed in [v0.16.1](https://github.com/terraform-providers/terraform-provider-tfe/blob/v0.16.1/CHANGELOG.md#0161-april-28-2020). Thanks for the report and PR!
